core/filter_api.php

```diff
@@ -1048,8 +1059,12 @@ function filter_get_field( $p_filter_id, $p_field_name ) {
  * @param boolean $p_show_sticky   Whether to show sticky items.
  * @param array   $p_query_clauses Array of query clauses.
  * @return array $p_query_clauses
+ *
+ * @deprecated	Use BugFilterQuery class
```

Comment:
We should not deprecate and generate trigger DEPRECATED errors as long as we use this function ourselves in function filter_get_bug_rows_query_clauses

Reply:
Hmm, filter_get_bug_rows_query_clauses is also deprecated and is not used anymore.

I am wondering if we could remove these functions instead of deprecation them.

Reply:
> I am wondering if we could remove these functions instead of deprecating them.

I have left them in, as an example: my own use case, i have plugins that use them.... so it's possible that others do it too.

Rewritting them, as envelopes for the new functionality, is not factible either (i think). Note that they work around queries and db_params, which have side effects in caller context. The caller may also have specific code to manage those side effects (Eg, db_param stack and query execution).

One thing which may be cleaner, is to move the deprecated code outside of filter_api?

Reply:
> I am wondering if we could remove these functions instead of deprecating them.

I have left them in, as an example: my own use case, i have plugins that use them.... so it's possible that others do it too.

Rewritting them, as envelopes for the new functionality, is not factible either (i think). Note that they work around queries and db_params, which have side effects in caller context. The caller may also have specific code to manage those side effects (Eg, db_param stack and query execution).

One thing which may be cleaner, is to move the deprecated code outside of filter_api?